site/gatsby-site/server/fields/entities.ts

```diff
@@ -1,28 +1,107 @@
-import { GraphQLFieldConfigMap, GraphQLObjectType } from "graphql";
+import { GraphQLFieldConfigMap, GraphQLInputObjectType, GraphQLList, GraphQLNonNull, GraphQLObjectType, GraphQLString } from "graphql";
 import { allow } from "graphql-shield";
 import { generateMutationFields, generateQueryFields } from "../utils";
 import { Context } from "../interfaces";
 import { EntityType } from "../types/entity";
-
+import { GraphQLDateTime, GraphQLJSONObject } from "graphql-scalars";
 
 export const queryFields: GraphQLFieldConfigMap<any, Context> = {
 
-    ...generateQueryFields({ collectionName: 'entities', Type: EntityType })
+  ...generateQueryFields({ collectionName: 'entities', Type: EntityType })
 }
 
+const UpdateOneEntityPayload = new GraphQLObjectType({
+  name: 'UpdateOneEntityPayload',
+  fields: {
+    created_at: { type: new GraphQLNonNull(GraphQLDateTime) },
+    date_modified: { type: new GraphQLNonNull(GraphQLDateTime) },
+    entity_id: { type: new GraphQLNonNull(GraphQLString) },
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    entity_relationships_to_add: { type: new GraphQLList(GraphQLString) },
+    entity_relationships_to_remove: { type: new GraphQLList(GraphQLString) },
+  },
+});
+
+const UpdateOneEntityInput = new GraphQLInputObjectType({
+  name: 'UpdateOneEntityInput',
+  fields: {
+    created_at: { type: new GraphQLNonNull(GraphQLDateTime) },
+    date_modified: { type: new GraphQLNonNull(GraphQLDateTime) },
+    entity_id: { type: new GraphQLNonNull(GraphQLString) },
+    name: { type: new GraphQLNonNull(GraphQLString) },
+    entity_relationships_to_add: { type: new GraphQLList(GraphQLJSONObject) },
+    entity_relationships_to_remove: { type: new GraphQLList(GraphQLJSONObject) },
+  },
+});
 
 export const mutationFields: GraphQLFieldConfigMap<any, Context> = {
 
-    ...generateMutationFields({ collectionName: 'entities', Type: EntityType, generateFields: ['updateOne', 'upsertOne'] }),
-}
+  ...generateMutationFields({ collectionName: 'entities', Type: EntityType, generateFields: ['updateOne', 'upsertOne'] }),
 
-export const permissions = {
-    Query: {
-        entity: allow,
-        entities: allow,
+  updateOneEntity: {
```

Comment:
I would use names for custom resolvers that do not coincide with generated ones to avoid confusion, but it is not critical.

Reply:
Also, please add an API test for this custom resolver. It should be in `server/tests/entities.ts`, (look at `server/tests/reports.ts` for inspiration)